client/client.go

```diff
@@ -488,7 +488,12 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		resp = rtresp.resp
 		switch {
 		case ctx.Err() != nil:
-			err = ctx.Err()
```

Comment:
I can see that happening-- the client uses the vendored context.Canceled but the context returns the non-vendored context.Canceled so `ctx.Err() == context.Canceled` may fail since they're different pointers.

I don't understand why `if err == ctx.Err()` won't work for checking if the error came from the context. Presumably the context should return the same pointer for the same error each time.